scipy/special/_basic.py

```diff
@@ -3413,3 +3414,48 @@ def zeta(x, q=None, out=None):
         return _ufuncs._riemann_zeta(x, out)
     else:
         return _ufuncs._zeta(x, q, out)
+
+      
+def softplus(x, *, out=None, where=True, **kwargs):
```

Comment:
```suggestion
def softplus(x, **kwargs):
```

@rgommers [you asked](https://github.com/scipy/scipy/pull/18605/files/4dd4ddb55d2f2c32a931c9b4e0943104135fb3f9#diff-9a3516307ecc6b8a41e8631035448716fb36d7433c46a55a46b6f6ebebdd1483) for this to be documented exactly as `logaddexp` so it would look like a ufunc. 

I've implemented that change, but it complicates things a tiny bit if we want to make it array-API compatible (https://github.com/scipy/scipy/pull/18605#issuecomment-2046598660) because not all backends will accept `out` and `where`. Shall we just accept `**kwargs` rather than `out` and `where` specifically and remove the specific documentation of `out` and `where` below, just referring users to the ufunc docs for kwargs?

Reply:
Personally, I'm fine with not putting `out` and `where` in the signature explicitly, and accepting only `**kwargs`, at least for now. Maybe there's something more that can be done down the line about `out` and `where` in the array API, but if so, I'm fine kicking that can down the road.